src/paths/widest_paths.c

```diff
@@ -0,0 +1,747 @@
+ *        \endclist
+ *
+ * Time complexity: O(|E|log|E|+|V|), where |V| is the number of
+ * vertices and |E| is the number of edges
+ *
+ * \sa \ref igraph_widest_paths_dijkstra() or \ref igraph_widest_paths_floyd_warshall()
+ * if you only need the path length but not the paths themselves.
+ */
+igraph_error_t igraph_get_widest_paths(const igraph_t *graph,
+                                       igraph_vector_ptr_t *vertices,
+                                       igraph_vector_ptr_t *edges,
+                                       igraph_integer_t from,
+                                       igraph_vs_t to,
+                                       const igraph_vector_t *weights,
+                                       igraph_neimode_t mode,
+                                       igraph_vector_int_t *predecessors,
+                                       igraph_vector_int_t *inbound_edges) {
+
+    /* Implementation details: This is a Dijkstra algorithm with a
+    binary heap, modified to support widest paths. The heap is indexed,
+    so it stores both the widest path to a node, as well as it's index. We
+    use a 2 way heap so that we can query indexes directly in the heap.
+
+    To adapt a Dijkstra to handle widest path, instead of prioritising candidate
+    nodes with the minimum distance, we prioritise those with the maximum
+    width instead. When adding a node into our set of 'completed' nodes, we
+    update all neighbouring nodes with a width that is equal to the min of the
+    width to the current node and the width of the edge.
+
+    We denote the widest path from a node to itself as infinity, and the widest
+    path from a node to a node it cannot reach as negative infinity.
+    */
+
+    igraph_integer_t no_of_nodes = igraph_vcount(graph);
+    igraph_integer_t no_of_edges = igraph_ecount(graph);
+    igraph_real_t my_posinfinity = IGRAPH_POSINFINITY;
+    igraph_real_t my_neginfinity = IGRAPH_NEGINFINITY;
+    igraph_vit_t vit;
+    igraph_2wheap_t Q;
+    igraph_lazy_inclist_t inclist;
+    igraph_vector_t widths;
+    igraph_integer_t *parents;
+    igraph_bool_t *is_target;
+    igraph_integer_t i, to_reach;
+
+    if (!weights) {
+        IGRAPH_ERROR("Weight vector is required.", IGRAPH_EINVAL);
+    }
+
+    if (igraph_vector_size(weights) != no_of_edges) {
+        IGRAPH_ERRORF("Weight vector length (%" IGRAPH_PRId ") does not match number "
+                      " of edges (%" IGRAPH_PRId ").", IGRAPH_EINVAL,
+                      igraph_vector_size(weights), no_of_edges);
+    }
+
+    if (igraph_vector_is_any_nan(weights)) {
+        IGRAPH_ERROR("Weight vector must not contain NaN values.", IGRAPH_EINVAL);
+    }
+
+    IGRAPH_CHECK(igraph_vit_create(graph, to, &vit));
+    IGRAPH_FINALLY(igraph_vit_destroy, &vit);
+
+    if (vertices && IGRAPH_VIT_SIZE(vit) != igraph_vector_ptr_size(vertices)) {
+        IGRAPH_ERROR("Size of `vertices' and `to' should match.", IGRAPH_EINVAL);
```

Comment:
Why do you give an error here? Shouldn't you resize `vertices`? That's what the docs suggest, and what for example `igraph_get_shortest_paths_dijkstra` also does.

Same goes for the `edges`.

Reply:
Genuinely not sure how that happened, since I would have had to manually change it to be something different! Either way, this is now fixed to resize `vertices` and `edges`, as noted by the documentation.